packages/components/src/data-store/hooks/use-user-vote.js

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+export default function useUserVote(props) {
+
+  let self = this;
+
+  const projectId = props.projectId || props.config.projectId;
+  const type = props.type || props.config.type;
+
+  const { data, error, isLoading } = self.useSWR({ type: 'user-vote', projectId: self.projectId }, 'userVote.fetch');
+
+  // add functionality
+  let userVote = data || {};
+  userVote.submitVote = function(vote) {
+    return self.mutate({ type: 'user-vote', projectId: self.projectId }, 'userVote.submitVote', vote, { action: 'update' });
+  }
+
+  if (error) {
+    let error = new Error(error);
+	  let event = new window.CustomEvent('osc-error', { detail: error });
```

Comment:
Nog nodig?

Reply:
Ik denk dat je hooks wilt creeren, misschien voor gebruik tussen losse widgets op een pagina, maar ook voor externe toepassingen. Dat is zeker nog geen defiinitief uitgewerkt idee. Maar ook hier: laten we dit buiten deze PR houden.